### PR TITLE
Travis Build Updates for mason-registry

### PR DESCRIPTION
**Pull request by oplambeck**
New script first detects what file is most recent, ie the most recent package to be added, then grabs it's source and attempts to clone the package. Then once the package is cloned it runs `mason publish --ci-test` which runs a series of compatibility checks to ensure that the package can be merged successfully with the mason-registry. 

Right now this pulls the master branch of chapel as that's where the `mason publish` commands are, but should probably be set to the 1.20 release when possible. 

As well as this PR relies on https://github.com/chapel-lang/chapel/pull/13938 being merged in order to work.

### CODE REVIEW

File: util/travisScript.bash [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+#Clones master of chapel and quickstarts with CHPL_REGEXP=re2
+git clone --depth=50 --branch=master https://github.com/chapel-lang/chapel.git
```
Comment (ben-albrecht) [inline util/travisScript.bash]: We might as well make the depth 1

File: util/travisScript.bash [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+cd chapel
+source util/quickstart/setchplenv.bash
+export CHPL_REGEXP=re2
+make -j 4
```
Comment (ben-albrecht) [inline util/travisScript.bash]: Do we know enough about the Travis hardware to know this will speed things up?

File: util/travisScript.bash [hunk 3]
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+#Clones master of chapel and quickstarts with CHPL_REGEXP=re2
```
Comment (ben-albrecht) [inline util/travisScript.bash]: nit: consistently add spaces before comments and consistently capitalize first word

```suggestion
# Clones master of chapel and quickstarts with CHPL_REGEXP=re2
```

File: util/travisScript.bash [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+git clone --depth=50 --branch=master https://github.com/chapel-lang/chapel.git
+cd chapel
+source util/quickstart/setchplenv.bash
+export CHPL_REGEXP=re2
+make -j 4
+
+#Runs a make check and makes mason
+make check
```
Comment (ben-albrecht) [inline util/travisScript.bash]: We should exit early if `make check` fails, right?

File: util/travisScript.bash [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+cd tools/mason
+make install
```
Comment (ben-albrecht) [inline util/travisScript.bash]: We should be able to just `make mason` from top-level.

File: util/travisScript.bash [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+source="$(grep source "$f" | cut -d= -f2)"
+echo $source
+#strips the quotes off of the source
+fixed=$(echo "$source" | tr -d '"')
+#clones the source
+echo $fixed
+git clone $fixed newPackage
+cd newPackage
+#runs mason publish --check --travis on the package
+mason publish --ci-check
```
Comment (ben-albrecht) [inline util/travisScript.bash]: How does Travis end up detecting the failure? If this returns a non-zero exit code? Or do we need to `exit $?` after this?

**Review by ben-albrecht — CHANGES_REQUESTED** [review]
Looking good. Have some suggested cleanups and suggested checks. Also, this might be more clean and maintainable if you consolidate some of the functionality into functions so that the main script looks something like:

```bash
git clone --depth=1 --branch=master https://github.com/chapel-lang/chapel.git
source chapel/util/quickstart/setchplenv.bash
export CHPL_REGEXP=re2

buildChapel

buildMason

package=getPackageName

checkPackage ${package}
```



Also, `rebase -i` this and get rid of all the noisy commits please :)

File: util/travisScript.bash (hunk 2 above)
Comment (oplambeck) [reply util/travisScript.bash]: Did a little digging online and it looks like this works. Also purely antedotal it seems to run quicker with that to me.

Comment (ben-albrecht) [reply util/travisScript.bash]: OK. I think it'll depend on how many cores are available on the Travis job and if it's running as a shared resource or not. Should be safe to leave in for now, and we can always remove it if we find it to be slow.

File: util/travisScript.bash [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  source util/quickstart/setchplenv.bash
+  export CHPL_REGEXP=re2
+  make
+ }
+
+# Runs a make check, and if it passes then makes mason
+makeCheckAndMason () {
+  make check
+  output=$?
+  if [[ '$output' -eq 0 ]]; then
```
Comment (ben-albrecht) [inline util/travisScript.bash]: I think this will work?

```suggestion
  if [ ${output} -eq 0 ]; then
```

**Review by ben-albrecht — APPROVED** [review]
LGTM - let's merge and do a test run with a dummy package.

**Pull request merged**